src/pages/instances/forms/ConfigurationTable.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, ReactNode, useEffect, useState } from "react";
+import { Icon, MainTable, Tooltip } from "@canonical/react-components";
+import {
+  MainTableCell,
+  MainTableHeader,
+  MainTableRow,
+} from "@canonical/react-components/dist/components/MainTable/MainTable";
+import { figureCollapsedScreen } from "util/formFields";
+import { SharedFormikTypes } from "pages/instances/forms/sharedFormTypes";
+import { EditInstanceFormValues } from "pages/instances/EditInstanceForm";
+import useEventListener from "@use-it/event-listener";
+import { updateTBodyHeight } from "util/updateTBodyHeight";
+import { useNotify } from "context/notify";
+
+interface Props {
+  formik: SharedFormikTypes;
+  rows: MainTableRow[];
+  configurationExtra?: ReactNode;
+  emptyStateMsg?: string;
+  isCollapsedOverride?: boolean;
+}
+
+const ConfigurationTable: FC<Props> = ({
+  formik,
+  rows,
+  configurationExtra,
+  emptyStateMsg,
+  isCollapsedOverride,
+}) => {
+  const notify = useNotify();
+  const [isSmallScreen, setSmallScreen] = useState(figureCollapsedScreen());
+  const isCollapsedView = isSmallScreen || isCollapsedOverride;
+
+  const resize = () => {
+    updateTBodyHeight("configuration-table");
+    setSmallScreen(figureCollapsedScreen());
+  };
+  useEventListener("resize", resize);
+  useEffect(resize, [notify.notification]);
+
+  const isReadOnly = (formik.values as EditInstanceFormValues).readOnly;
+
+  const hasOverrideClass = (item: MainTableHeader | MainTableCell) =>
+    item.className !== "override";
+
+  const filterHeaders = (headers: MainTableHeader[]): MainTableHeader[] => {
+    const prefiltered = isCollapsedView
+      ? headers.filter((header) => header.className !== "value")
+      : headers;
+    return isReadOnly ? prefiltered.filter(hasOverrideClass) : prefiltered;
+  };
+
+  const filterRows = (rows: MainTableRow[]): MainTableRow[] => {
+    return isReadOnly
+      ? rows.map((row) => {
+          const filteredColumns = row.columns?.filter(hasOverrideClass);
+          return { ...row, columns: filteredColumns };
+        })
+      : rows;
+  };
+
+  const collapseRows = (rows: MainTableRow[]): MainTableRow[] => {
+    return isCollapsedView
+      ? rows.map((row) => {
+          const collapsedColumns = [
+            row.columns?.find((col) => col.className === "override") ?? {},
+            {
+              content: (
+                <>
+                  {row.columns?.find((col) => col.className === "config")
+                    ?.content ?? ""}
```

Comment:
Wrapping it here, when it gets collapsed makes more sense, yes. Should all work now as expected. Also simplified the labels.

Reply:
I think you should also remove all the `<div>` wrappers that you set on the child components, in order to avoid polluting the DOM like this:

![image](https://user-images.githubusercontent.com/56583786/228289692-ece689f9-3917-404e-855e-c914e84486d9.png)